sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared/uploads.py

```diff
@@ -477,6 +477,13 @@ def read(self, size=None):
                             raise IOError("Stream failed to seek to the desired location.")
```

Comment:
Why do we fail fast in the case of a lock instead of resetting to absolute position like what you've added? I'm guessing this could potentially mess with where other threads should reposition?

Reply:
we've already seeked and cannot find the absolute position
```
                        self._wrapped_stream.seek(absolute_position, SEEK_SET)
                        # If we can't seek to the right location, our read will be corrupted so fail fast.
                        if self._wrapped_stream.tell() != absolute_position:
                            raise IOError("Stream failed to seek to the desired location.")
```